File: app/src/main/java/com/yl/study/MyApplication.java
package com.yl.study;

import android.app.Application;
import android.content.Context;
import android.os.Bundle;
import android.os.Debug;
import android.os.Environment;

import java.io.File;

/**
 * @date: $ $
 * @author: yanglei
 * @description
 */
public class MyApplication extends Application {
    public MyApplication() {
        // 统计耗时。文件名；文件保存大小；捕捉频率，
//        Debug.startMethodTracingSampling(new File(Environment.getExternalStorageDirectory(),
//                "study_tracing_log").getAbsolutePath(), 8 * 1024 * 1024, 1_000);
    }
    private static Context context;
    @Override
    public void onCreate() {
        super.onCreate();
        context = this;
//        context = getApplicationContext();
//        context = getBaseContext();
    }

    public static Context getContext() {
        return context;
    }
}

File: app/src/main/java/com/yl/study/MyApplication.kt
package com.yl.study

import android.content.Context
import android.util.Log
import com.yl.base.BaseApplication

/**
 * @date: $ $
 * @author: yanglei
 * @description
 */
//@HiltAndroidApp
public class MyApplication : BaseApplication() {
    override fun attachBaseContext(base: Context?) {
        super.attachBaseContext(base)
        Log.d("tag-","==========MyApplication attachBaseContext==============="+System.currentTimeMillis())
    }
    override fun onCreate() {
        super.onCreate()
        Log.d("tag-","==========MyApplication onCreate()============="+System.currentTimeMillis())

    }
}